### PR TITLE
gRPC route to use mainrouter when http.root-path is not '/'

### LINKED ISSUE #34261: grpc service not working when use-separate-server is set to false and custom http.root-path is set
### Describe the bug

When quarkus.grpc.server.use-separate-server=false and quarkus.http.root-path=api/v1 is set, the grpc service is not working. On removing the http root path it works. Is there any specific that needs to be added for the grpc to recognize this custom root path.

### Expected behavior

grpc service should be accessible even with custom http root path

### Actual behavior

grpc service gives 404 error when accessed

### How to Reproduce?

Steps to reproduce

- Take the grpc plaintext quickstart
- Add both use-separate-server false and custom http.root-path
- From Dev UI try to invoke the grpc service

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.16.7.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by SivaM07**
Fixes https://github.com/quarkusio/quarkus/issues/34261

### CODE REVIEW

File: extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/VertxHttpProcessor.java [hunk 1]
```diff
@@ -278,6 +278,14 @@ VertxWebRouterBuildItem initializeRouter(VertxHttpRecorder recorder,
             }
         }
 
+        /**
+         * To create mainrouter when `${quarkus.http.root-path}` is not {@literal /}
+         * Refer https://github.com/quarkusio/quarkus/issues/34261
+         */
+        if (!httpBuildTimeConfig.rootPath.equals("/") && !mainRouterCreated) {
+            mainRouter = recorder.initializeRouter(vertx.getVertx());
```
Comment (cescoffier) [inline extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/VertxHttpProcessor.java:286]: We need to be sure it's used correctly.

**Review by alesj — APPROVED** [review]

**Comment by alesj** [comment]
Would be also good to have that @SivaM07 reproducer / tests,
but if everything from before passes, I guess we're good to go,
and we can add those tests later ...

File: extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/VertxHttpProcessor.java (hunk 1 above)
Comment (SivaM07) [reply extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/VertxHttpProcessor.java:286]: Is there any specific usage you are concerned about ?

**Comment by SivaM07** [comment]
@cescoffier @alesj , both the failed check says just cancelled, is it possible someone cancelled it ? could you please guide on this ?

**Comment by SivaM07** [comment]
> I would prefer having tests in this PR. Just to be sure it actually does the job.

@cescoffier Added testcase to show that post fix the combination of separate server set to false and custom http.root-path works.

File: extensions/grpc/deployment/src/test/java/io/quarkus/grpc/server/GrpcCustomHttpRootPathTest.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package io.quarkus.grpc.server;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.grpc.examples.helloworld.*;
```
Comment (cescoffier) [inline extensions/grpc/deployment/src/test/java/io/quarkus/grpc/server/GrpcCustomHttpRootPathTest.java:10]: Can you use single import statements?

File: extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/VertxHttpProcessor.java (hunk 1 above)
Comment (cescoffier) [reply extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/VertxHttpProcessor.java:286]: The test covers it. Let's see what the CI says.

**Review by cescoffier — APPROVED** [review]

**Comment by cescoffier** [comment]
Can you fix the * import? Then, I think we will be good to go.

File: extensions/grpc/deployment/src/test/java/io/quarkus/grpc/server/GrpcCustomHttpRootPathTest.java (hunk 2 above)
Comment (SivaM07) [reply extensions/grpc/deployment/src/test/java/io/quarkus/grpc/server/GrpcCustomHttpRootPathTest.java]: Fixed, please check now.

**Comment by SivaM07** [comment]
> Can you fix the * import? Then, I think we will be good to go.

@cescoffier thanks for approving, the import statements are fixed now, please check.

**Comment by cescoffier** [comment]
Thanks!

**Comment by SivaM07** [comment]
@cescoffier can we get this fix into 2.16.x release too ?

**Comment by cescoffier** [comment]
It should be backportable, but I let @gsmet decides.

**Comment by gsmet** [comment]
I backported it to 2.16.8.Final that will be released tomorrow.

**Pull request merged**